packages/wdio-sauce-service/src/service.ts

```diff
@@ -216,17 +219,39 @@ export default class SauceService implements Services.ServiceInstance {
 
         const status = 'status: ' + (failures > 0 ? 'failing' : 'passing')
         if (!this._browser.isMultiremote) {
+            await this._uploadLogs(this._browser.sessionId)
             log.info(`Update job with sessionId ${this._browser.sessionId}, ${status}`)
             return this._isUP ? this.updateUP(failures) : this.updateJob(this._browser.sessionId, failures)
         }
 
         const mulitremoteBrowser = this._browser as MultiRemoteBrowser<'async'>
-        return Promise.all(Object.keys(this._capabilities).map((browserName) => {
+        return Promise.all(Object.keys(this._capabilities).map(async (browserName) => {
+            await this._uploadLogs(mulitremoteBrowser[browserName].sessionId)
             log.info(`Update multiremote job for browser "${browserName}" and sessionId ${mulitremoteBrowser[browserName].sessionId}, ${status}`)
             return this._isUP ? this.updateUP(failures) : this.updateJob(mulitremoteBrowser[browserName].sessionId, failures, false, browserName)
         }))
     }
 
+    /**
+     * upload files to Sauce Labs platform
+     * @param jobId id of the job
+     * @returns a promise that is resolved once all files got uploaded
+     */
+    private async _uploadLogs (jobId: string) {
+        if (!this._options.uploadLogs || !this._config.outputDir) {
+            return
+        }
+
+        const files = (await fs.promises.readdir(this._config.outputDir))
```

Comment:
Am I reading this correctly that this is connected to also writing the logs to your local disk in order for this to work?
If so, I'm not seeing that back in the docs, secondly, can we read this from the stream by default and push it then to Sauce, so the local disk won't be "polluted" with extra files?

Reply:
Good point, let me look into this. My hunch is that it would only work for logs we store in files because a stream never keeps data in memory.

Reply:
Updated readme to add option description there. Unfortunately it is not possible to get data from a stream once written to the target (here stdout).